### PR TITLE
Fix ApolloClient authorization header during SSR

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

SSR ApolloClient never sent authentication header, fixed this

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1525** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1525" target="_blank">Graphite</a> 
    * **PR #1529** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1529" target="_blank">Graphite</a> 
      * **PR #1558** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1558" target="_blank">Graphite</a>  👈
      * **PR #1549** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1549" target="_blank">Graphite</a> 
        * **PR #1554** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1554" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1558?utm_source=stack-comment).

**Review by robinandeer — COMMENTED** [review]
Looks good to me, but I don't understand how this has anything to do with SSR 🤔

The only place in the diff where we are fetching auth headers is if `isBrowser()`, no?

File: apps/store/src/services/apollo/client.ts [hunk 1]
```diff
@@ -61,7 +65,6 @@ const createApolloClient = (headers?: Record<string, string>) => {
         },
       },
     }),
-    headers,
     connectToDevTools: process.env.NODE_ENV === 'development',
   })
 }
```
Comment (alebedev) [inline apps/store/src/services/apollo/client.ts:70]: <i class='graphite__hidden'>[Re: lines 83 to 83]<br/></i>

@robinandeer we build SSR headers here.  The problem was that we were overwriting them with empty object with `authLink`

I'm not particularly happy with how it works now - SSR and client-side auth are separated - would be nice to refactor it later in a way that puts all auth-related code together<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1558?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by robinandeer — APPROVED** [review]

**Comment by alebedev** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: LNt7OB37uAsddXEUWHM6.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1558)

**Pull request merged**